yunwoo/LV_1/기사단원의-무기.js

```diff
@@ -0,0 +1,18 @@
+function solution(number, limit, power) {
+  const divisors = Array.from({ length: number }, (_, i) => getDivisor(i + 1));
+  return divisors.reduce((acc, cur) => (acc += cur > limit ? power : cur), 0);
+}
+
+function getDivisor(number) {
+  let result = 0;
+
+  for (let i = 1; i <= Math.sqrt(number); i++) {
+    if (number % i === 0) {
+      result++;
+
+      if (i * i < number) result++;
+    }
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(number, limit, power) {
  const divisors = Array.from({ length: number }, (_, i) => getDivisor(i + 1));
  return divisors.reduce((acc, cur) => (acc += cur > limit ? power : cur), 0);
}

function getDivisor(number) {
  let result = 0;

  for (let i = 1; i <= Math.sqrt(number); i++) {
    if (number % i === 0) {
      result++;

      if (i * i < number) result++;
    }
  }

  return result;
}
```
새롭게 함수 선언을 해서 코드가 더 눈에 잘 들어오는 것 같아요! 윤우님 코드로 새로운 문제풀이방법 익혀갑니다~!

Reply:
- 👍 함수를 분리하는것도 좋은 방법이죠ㅎㅎ